crates/autopilot/src/database/orders.rs

```diff
@@ -62,38 +70,52 @@ impl Postgres {
         &self,
         order_uid: &OrderUid,
         update: &FeeUpdate,
-        quote: &LimitOrderQuote,
     ) -> Result<()> {
+        let (update, quote) = match update {
+            FeeUpdate::Success {
+                timestamp,
+                surplus_fee,
+                full_fee_amount,
+                quote,
+            } => (
+                database::orders::FeeUpdate {
+                    surplus_fee: Some(u256_to_big_decimal(surplus_fee)),
+                    surplus_fee_timestamp: *timestamp,
+                    full_fee_amount: u256_to_big_decimal(full_fee_amount),
+                },
+                Some(database::orders::Quote {
+                    order_uid: ByteArray(order_uid.0),
+                    gas_amount: quote.fee_parameters.gas_amount,
+                    gas_price: quote.fee_parameters.gas_price,
+                    sell_token_price: quote.fee_parameters.sell_token_price,
+                    sell_amount: u256_to_big_decimal(&quote.sell_amount),
+                    buy_amount: u256_to_big_decimal(&quote.buy_amount),
+                }),
+            ),
+            FeeUpdate::Failure { timestamp } => (
+                database::orders::FeeUpdate {
+                    surplus_fee: None,
```

Comment:
This seems a bit risky. With the timestamp now being updates `surplus_fee == None` is the only thing preventing this limit order from being part of the solvable orders.
Maybe we could add a comment here saying that `surplus_fee` absolutely has to stay `None` in this case.

Reply:
Added comment